src/lib/routes/admin-api/tag.ts

```diff
@@ -19,42 +23,127 @@ class TagController extends Controller {
 
-        this.get('/', this.getTags);
-        this.post('/', this.createTag, UPDATE_FEATURE);
-        this.get('/:type', this.getTagsByType);
-        this.get('/:type/:value', this.getTag);
-        this.delete('/:type/:value', this.deleteTag, UPDATE_FEATURE);
+        this.route({
+            method: 'get',
+            path: '',
+            handler: this.getTags,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'getTags',
+                    responses: { 200: createResponseSchema('tagsSchema') },
+                }),
+            ],
+        });
+        this.route({
+            method: 'post',
+            path: '',
+            handler: this.createTag,
+            permission: UPDATE_FEATURE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'createTag',
+                    responses: {
+                        201: emptyResponse,
+                    },
+                    requestBody: createRequestSchema('tagSchema'),
+                }),
+            ],
+        });
+        this.route({
+            method: 'get',
+            path: '/:type',
+            handler: this.getTagsByType,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'getTagsByType',
+                    responses: {
+                        200: createResponseSchema('tagsSchema'),
+                    },
+                }),
+            ],
+        });
+        this.route({
+            method: 'get',
+            path: '/:type/:value',
+            handler: this.getTag,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'getTag',
+                    responses: {
+                        200: createResponseSchema('tagSchema'),
+                    },
+                }),
+            ],
+        });
+        this.route({
+            method: 'delete',
+            path: '/:type/:value',
+            handler: this.deleteTag,
+            acceptAnyContentType: true,
+            permission: UPDATE_FEATURE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'deleteTag',
+                    responses: {
+                        200: emptyResponse,
+                    },
+                }),
+            ],
+        });
     }
 
-    async getTags(req: Request, res: Response): Promise<void> {
+    async getTags(req: Request, res: Response<TagsSchema>): Promise<void> {
         const tags = await this.tagService.getTags();
         res.json({ version, tags });
```

Comment:
Could use `respondWithValidation` here and in the other getters.

Reply:
Good point. I'll update :)